src/main/java/io/cdap/plugin/gcp/spanner/common/SpannerUtil.java

```diff
@@ -39,12 +56,117 @@
+                                                               boolean isServiceAccountFilePath,
+                                                               String projectId,
+                                                               Counter counter) throws IOException {
+      class InterceptedClientCall<ReqT, RespT> extends ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT> {
+        InterceptedClientCall(ClientCall<ReqT, RespT> call) {
+          super(call);
+        }
+
+        @Override
+        public void start(Listener<RespT> responseListener, Metadata headers) {
+          super.start(new ForwardingClientCallListener.
+            SimpleForwardingClientCallListener<RespT>(responseListener) {
+            @Override
+            public void onMessage(RespT message) {
+              if (message instanceof PartialResultSet) {
+                PartialResultSet partialResultSet = (PartialResultSet) message;
+                counter.increment(partialResultSet.getSerializedSize());
+              } else if (message instanceof ResultSet) {
+                ResultSet resultSet = (ResultSet) message;
+                counter.increment(resultSet.getSerializedSize());
+              }
+              super.onMessage(message);
+            }
+          }, headers);
+        }
+      }
+
+      SpannerOptions.Builder optionsBuilder = buildSpannerOptions(serviceAccount, isServiceAccountFilePath, projectId);
+      optionsBuilder.setInterceptorProvider(
+      SpannerInterceptorProvider.createDefault()
+      .with(
+        new ClientInterceptor() {
+          @Override
+          public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+            MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+            ClientCall<ReqT, RespT> call = next.newCall(method, callOptions);
+            return new InterceptedClientCall<>(call);
+          }
+        }
+      ));
+
+    return optionsBuilder.build().getService();
+  }
+
+  /**
+   * Construct and return the {@link Spanner} service with an interceptor that increments the provided counter by the
+   * number of bytes written to Spanner.
+   */
+  public static Spanner getSpannerServiceWithWriteInterceptor(String serviceAccount,
+                                                              boolean isServiceAccountFilePath,
+                                                              String projectId,
+                                                              Counter counter) throws IOException {
+    class InterceptedClientCall<ReqT, RespT> extends ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT> {
+      InterceptedClientCall(ClientCall<ReqT, RespT> call) {
+        super(call);
+      }
+      @Override
+      public void sendMessage(ReqT message) {
+        if (message instanceof CommitRequest) {
+          CommitRequest commitRequest = (CommitRequest) message;
+          commitRequest.getMutationsList().stream()
```

Comment:
Add a small comment on what this does, not immediately obvious

Reply:
added comment.